src/main/asciidoc/transactions.adoc

```diff
@@ -33,82 +41,257 @@ static class Config extends AbstractCouchbaseConfiguration {
+will roll back on RuntimeException and Error but not on checked
+exceptions.
+ 
+Rollback rules determine if a transaction should be rolled back when a given
+exception is thrown, and the rules are based on types or patterns. Custom
+rules may be configured via rollbackFor/noRollbackFor and
+#rollbackForClassName/noRollbackForClassName, which allow
+rules to be specified as types or patterns, respectively.
+ 
+When a rollback rule is defined with an exception type, that type will be
+used to match against the type of a thrown exception and its super types,
+providing type safety and avoiding any unintentional matches that may occur
+when using a pattern. For example, a value of
+jakarta.servlet.ServletException.class will only match thrown exceptions
+of type jakarta.servlet.ServletException and its subclasses.
+ 
+When a rollback rule is defined with an exception pattern, the pattern can
+be a fully qualified class name or a substring of a fully qualified class name
+for an exception type (which must be a subclass of Throwable), with no
+wildcard support at present. For example, a value of
+"jakarta.servlet.ServletException" or "ServletException" will
+match jakarta.servlet.ServletException and its ubclasses.
+ 
+WARNING: You must carefully consider how specific a pattern
+is and whether to include package information (which isn't mandatory). For example,
+"Exception" will match nearly anything and will probably hide other
+rules. "java.lang.Exception" would be correct if "Exception"
+were meant to define a rule for all checked exceptions. With more unique
+exception names such as "BaseBusinessException" there is likely no
+need to use the fully qualified class name for the exception pattern. Furthermore,
+rollback rules defined via patterns may result in unintentional matches for
+similarly named exceptions and nested classes. This is due to the fact that a
+thrown exception is considered to be a match for a given pattern-based rollback
+rule if the name of thrown exception contains the exception pattern configured
+for the rollback rule. For example, given a rule configured to match against
+"com.example.CustomException", that rule will match against an exception
+named com.example.CustomExceptionV2 (an exception in the same package as
+CustomException but with an additional suffix) or an exception named
+com.example.CustomException$AnotherException (an exception declared as
+a nested class in CustomException).
+ 
+For specific information about the semantics of other attributes in this
+annotation, consult the org.springframework.transaction.TransactionDefinition
+and org.springframework.transaction.interceptor.TransactionAttribute javadocs.
+ 
+=== Transaction Management
+ 
+This annotation commonly works with thread-bound transactions managed by a
+org.springframework.transaction.PlatformTransactionManager, exposing a
+transaction to all data access operations within the current execution thread.
```

Comment:
... not even sure what this means?

This block is quite a bit of complexity to throw at a user.  Would a simple example not better demo things?

Reply:
For my reference, this is now resolved.